PXC-4158: Fix compilation issues with Azure Pipelines

inikep opened the pull request:
1. Disable TokuDB and RocksDB storage engines.
2. Remove macOS testing.
3. Fix gcc/clang compilation issues.

it-percona-cla commented:
[![CLA assistant check](https://cla-assistant.percona.com:443/pull/badge/signed)](https://cla-assistant.percona.com:443/percona/percona-xtradb-cluster?pullRequest=1744) <br/>All committers have signed the CLA.

inikep committed:
PXC-4158: Fix compilation issues with Azure Pipelines

1. Disable TokuDB and RocksDB storage engines.
2. Remove macOS testing.
3. Fix gcc/clang compilation issues.

venkatesh-prasad-v commented on azure-pipelines.yml:
```diff
@@ -335,28 +393,41 @@ jobs:
+        -DCMAKE_CXX_COMPILER=$SELECTED_CXX
+        -DCMAKE_C_COMPILER_LAUNCHER=ccache
+        -DCMAKE_CXX_COMPILER_LAUNCHER=ccache
+        -DWITH_COREDUMPER=ON
```
PXC will be built only when `-DWITH_WSREP=ON` is enabled. So, please enable `-DWITH_WSREP=ON`.

dutow commented:
Please also test this based on the current release-8.0.31 release branch. The 8.0 branch doesn't built galera with cmake yet, which can cause additional failures.
